_posts/2022-08-29-sbt-tutorial.md

```diff
@@ -0,0 +1,514 @@
+```
+test:testOnly *SimpleTest
+```
+This will run all the classes that ends with _SimpleTest_ within the entire project irrespective of the package.
+
+So far, we started the session and then ran the test. We can do that directly from the terminal by using the command:
+```
+sbt test 
+```
+This will start the sbt session and then run the test. However, we need to wrap the arguments after _sbt_ if there are some special characters are involved. For example, to run the test using wildcard, we need to use the command:
+```
+sbt 'test:testOnly *SimpleTest'
+```
+We can also use double quotes(") instead of single quotes('). This will ensure that all the arguments are passed to the sbt session at once.
+
+We can compile only the test classes by using the command `test:compile` within sbt session. 
+
+## Advanced configuration in build.sbt
+In the previous section, we saw configurations like _name_, _organization_, etc. The values we set there are applicable for the entire project. SBT also let's us to do some configurations per scope. 
+For example, by default sbt runs all the tests in parallel. If we somehow want to avoid that and run the tests in sequential way, we can set the config option for that. SBT uses the configuration key _parallelExecution_ for that and it is under the scope of Test. 
+
+So, in build.sbt, we can use the setting as:
+```
+Test / parallelExecution := false
+```
+This will set the value to false within the scope of test.
+
+There are many such configurations which are configured using the scope. You can think of this something in the lines of _Test.parallelExecution_.
+
+## Multi Module Project
+So far we have used a single module project. SBT allows to create multi-module project. This way, we can clearly separate different modules, but we can aggregate and combine different modules together to make a big project. Let's create a simple multi-module project first. For that, we can create a build.sbt file as below within a new empty directory.
+
+```
+ThisBuild / scalaVersion := "2.13.8"
+ThisBuild / version := "1.0"
+ThisBuild / name := "multi-module"
+ThisBuild / organization := "com.rockthejvm"
+
+lazy val module_1 = (project in file("module-1"))
+lazy val module_2 = (project in file("module-2"))
+```
+Now, when we can save this file and hit _sbt_ command in the project directoy. This will import the project based on the build.sbt we created and will also create 2 sub directores as _module-1_ and _module-2_ within the directory. The value provided in _file()_ is used to create the sub module name. 
+
+However, as of now there is no relationship between any of the modules. We can explicitly combine both the sub modules together and link to the parent project by adding a new line as belwo to the build.sbt:
+```
+lazy val root = (project in file("."))
+  .aggregate(util, core)
+```
+
+In the same way, we can define dependency between multiple submodules. We can make module-2 to depend on module-1 using dependsOn method:
```

Comment:
- dependency -> "a dependency" OR "dependencies"
- use the module names and the method in between ``